### PR TITLE
Fixed backspace bug on empty inputcommand. Added Autocomplete.

### PR DESCRIPTION
**Pull request by burcelee**

### CODE REVIEW

**Review by rebelmuffin — APPROVED** [review]
good work

File: Runtime/CheatConsole.cs [hunk 1]
```diff
@@ -109,6 +110,28 @@ private void UpdateInputField()
             m_inputCommand = m_inputCommand.Remove(m_cursorIndex, 1);
             m_dirty = true;
         }
+        else if (Input.GetKeyDown(KeyCode.Tab))
+        {
+            string[] all_method_names = CheatManager.GetAllMethodNames();
+            List<string> matching_names = new(all_method_names.Where(s => s.Contains(m_inputCommand)));
```
Comment (BenPyton) [inline Runtime/CheatConsole.cs]: You should use `camelCase` on your variable names for consistency with the rest of the code and Unity.

File: Runtime/CheatConsole.cs [hunk 2]
```diff
@@ -120,11 +143,15 @@ private void UpdateInputField()
...
-                    m_inputCommand = m_inputCommand.Remove(m_cursorIndex - 1, 1);
-                    m_cursorIndex--;
-                    m_dirty = true;
+                    if (m_cursorIndex > 0)
+                    {
+                        m_inputCommand = m_inputCommand.Remove(m_cursorIndex - 1, 1);
+                        m_cursorIndex--;
+                        m_dirty = true;
+                    }
+
```
Comment (BenPyton) [inline Runtime/CheatConsole.cs]: Maybe you could do a separate commit for this fix

**Comment by BenPyton** [comment]
Great job!
Thank you for contributing!

I have added comments on your commit to improve it a little before merging.

Moreover, maybe we could apply the same fix of the backspace but also for the delete key?
I don't know if there is any issue when pressing delete key if `inputCommand` is empty...

File: Runtime/CheatConsole.cs [hunk 3]
```diff
@@ -109,6 +110,28 @@ private void UpdateInputField()
...
+            }
+            else if (matching_names.Count == 1)
+            {
+                m_inputCommand = matching_names[0];
+                m_cursorIndex = m_inputCommand.Count();
+                m_dirty = true;
+            }
+            else
+            {
+                foreach (string method_name in matching_names)
```
Comment (BenPyton) [inline Runtime/CheatConsole.cs]: Same here: it would be better to use `camelCase` instead of `snake_case` for consistency.

**Comment by burcelee** [comment]
I'm not sure I can find the comments? I'm a little unfamiliar with the GitHub interface though, so maybe I'm just failing.

I can take a look at delete. I think I also found a bug when pressing up before there are any previous commands. I can potentially add in a fix for that as well.

**Comment by BenPyton** [comment]
> I'm not sure I can find the comments? I'm a little unfamiliar with the GitHub interface though, so maybe I'm just failing.

@burcelee Okay it was my fault, i'm not very familiar neither with github pull request features...
I've done it on my phone and I didn't saw the review submitting button... 😅

**Comment by burcelee** [comment]
Alright. I did an update with some split up commits. It's a bit of a pain to split commits since I'm using this plugin in my own repo and need to recreate the changes a little at a time in a fork of the cheatconsole repo. I do notice there is now a conflict, although I frankly don't understand how that happened or the nature of the conflict. Hope that helps...

**Comment by BenPyton** [comment]
Hi @burcelee,
After you made a pull request, I've updated the repo settings to allow only rebase merging of pull requests, so git history will remain linear. (I didn't set up correctly that on my old repos like this one)

I can see from the network graph that your fork is not updated on the last commit of my repo (where I've added the delete key support).
image

To resolve your conflict you should update your repo by rebasing your branch on my last commit.
***You can do it on a new clone of your repo if you don't want to mess something in your project!***
On github, it is not really user friendly to do, but you can follow [these steps](https://stackoverflow.com/a/7244456) to do it.

When doing the rebase, you should get this message:
```
$ git rebase upstream/master
First, rewinding head to replay your work on top of it...
Applying: Adding case insensitive autocomplete.
Using index info to reconstruct a base tree...
M       Runtime/CheatConsole.cs
Falling back to patching base and 3-way merge...
Auto-merging Runtime/CheatConsole.cs
CONFLICT (content): Merge conflict in Runtime/CheatConsole.cs
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch' to see the failed patch
Patch failed at 0001 Adding case insensitive autocomplete.

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

You have to fix manually the conflit because my last commit had changed the `CheatConsole.cs` too.
In a text editor, you just have to remove the lines starting with `>>>>>>>`, `=======` and `<<<<<<<`.

Then you enter the commands:
```
git add .
git rebase --continue
```

After that all are done! You just have to force push the rebased master branch on your github repo.

**Comment by burcelee** [comment]
Thanks for the excellent tutorial! Sorry about the messed up initial fork - honestly not totally sure how that happened. Hope everything is set now.

**Comment by BenPyton** [comment]
It's all good now! I can merge your pull request! 🎉 
Thank you for your contribution! 😄
